skyportal/handlers/api/user_obj_list.py

```diff
@@ -0,0 +1,356 @@
+        ---
+        description: Add a listing, if it doesn't exist yet
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  user_ud:
+                    type: integer
+                    required: true
+                  obj_id:
+                    type: string
+                    required: true
+                  list_name:
+                    type: string
+                    required: true
+                    description: |
+                        Listing name for this item, e.g., "favorites".
+                        Multiple objects can be saved by the same user to different
+                        lists, where the list names are user-defined.
+                        List name must be a non-empty string starting with an
+                        alphanumeric character or underscore.
+                        (it must match the regex: /^\\w+/)
+
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            listing_id:
+                              type: integer
+                              description: New listing ID
+
+        """
+
+        return self.add_listing(error_if_exists=False)
+
+    @auth_or_token
+    def patch(self, listing_id):
+        """
+        ---
+        description: Update a listing with new parameters
+        parameters:
+        - in: path
+          name: listing_id
+          required: true
+          schema:
+            type: integer
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  user_ud:
+                    type: integer
+                    required: true
+                  obj_id:
+                    type: string
+                    required: true
+                  list_name:
+                    type: string
+                    required: true
+                    description: |
+                        Listing name for this item, e.g., "favorites".
+                        Multiple objects can be saved by the same user to different
+                        lists, where the list names are user-defined.
+                        List name must be a non-empty string starting with an
+                        alphanumeric character or underscore.
+                        (it must match the regex: /^\\w+/)
+
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+
+        """
+
+        listing = Listing.query.get(int(listing_id))
```

Comment:
Handle any exceptions that may result from `listing_id` not being `int`-coercable